### PR TITLE
use alpaka 0.5.0-dev

### PR DESCRIPTION
**Pull request by psychocoderHPC**
- update CMake to support alpaka 0.5.0-dev
- C++14 or newer is now required
- CMake 3.15.0+ is now required
- update [alpaka to 0.5.0-dev](https://github.com/ComputationalRadiationPhysics/alpaka/commit/aa98e2b5914bcc41d0311954cbc74a75bb032e2b)
```
GIT_AUTHOR_NAME="Third Party" GIT_AUTHOR_EMAIL="cupla@hzdr.de" git subtree pull --prefix alpaka https://github.com/ComputationalRadiationPhysics/alpaka.git develop --squash
```

- [x] requires the alpaka fix  https://github.com/ComputationalRadiationPhysics/alpaka/pull/950
- [x] please merge after #163

### CODE REVIEW

File: cuplaConfig.cmake [hunk 1]
```diff
@@ -114,35 +114,18 @@ set_property(CACHE cupla_ALPAKA_PROVIDER PROPERTY STRINGS "intern;extern")
...
+    set(BUILD_TESTING OFF)
+    add_subdirectory(${_cupla_ROOT_DIR}/alpaka ${CMAKE_BINARY_DIR}/alpaka)
 else()
     find_package(alpaka HINTS $ENV{ALPAKA_ROOT})
 endif()
 
-if(NOT alpaka_FOUND)
-    message(WARNING "Required cupla dependency alpaka could not be found!")
+if(NOT TARGET alpaka::alpaka)
+    message(WARNING "Required cupla dependency alpaka could not be found! ")
```
Comment (sbastrakov) [inline cuplaConfig.cmake]: Do we need the added extra space at the end?

File: doc/PortingGuide.md [hunk 2]
```diff
@@ -2,7 +2,7 @@ Requirements to Port Your Project to *cupla*
 ============================================
 
 - your build system must be `CMake`
```
Comment (sbastrakov) [inline doc/PortingGuide.md]: Unrelated to this PR, but I think this is not required anymore.

Comment (psychocoderHPC) [reply doc/PortingGuide.md]: Yes should be fixed separate

**Comment by psychocoderHPC** [comment]
ready to merge

**Comment by psychocoderHPC** [comment]
please merge after #163

**Comment by psychocoderHPC** [comment]
yes this can be merged

**Pull request merged**